src/test/kotlin/org/rust/cargo/util/CargoWithToolchainCommandCompletionProviderTest.kt

```diff
@@ -0,0 +1,205 @@
+            "sparc64-unknown-linux-gnu",
+            "sparc64-unknown-netbsd",
+            "sparc64-unknown-openbsd",
+            "sparcv9-sun-solaris",
+            "thumbv4t-none-eabi",
+            "thumbv6m-none-eabi",
+            "thumbv7a-pc-windows-msvc",
+            "thumbv7a-uwp-windows-msvc",
+            "thumbv7em-none-eabi",
+            "thumbv7em-none-eabihf",
+            "thumbv7m-none-eabi",
+            "thumbv7neon-linux-androideabi",
+            "thumbv7neon-unknown-linux-gnueabihf",
+            "thumbv7neon-unknown-linux-musleabihf",
+            "thumbv8m.base-none-eabi",
+            "thumbv8m.main-none-eabi",
+            "thumbv8m.main-none-eabihf",
+            "wasm32-unknown-emscripten",
+            "wasm32-unknown-unknown",
+            "wasm32-wasi",
+            "x86_64-apple-darwin",
+            "x86_64-apple-ios",
+            "x86_64-apple-ios-macabi",
+            "x86_64-apple-tvos",
+            "x86_64-fortanix-unknown-sgx",
+            "x86_64-fuchsia",
+            "x86_64-linux-android",
+            "x86_64-linux-kernel",
+            "x86_64-pc-solaris",
+            "x86_64-pc-windows-gnu",
+            "x86_64-pc-windows-msvc",
+            "x86_64-rumprun-netbsd",
+            "x86_64-sun-solaris",
+            "x86_64-unknown-dragonfly",
+            "x86_64-unknown-freebsd",
+            "x86_64-unknown-haiku",
+            "x86_64-unknown-hermit",
+            "x86_64-unknown-hermit-kernel",
+            "x86_64-unknown-illumos",
+            "x86_64-unknown-l4re-uclibc",
+            "x86_64-unknown-linux-gnu",
+            "x86_64-unknown-linux-gnux32",
+            "x86_64-unknown-linux-musl",
+            "x86_64-unknown-netbsd",
+            "x86_64-unknown-openbsd",
+            "x86_64-unknown-redox",
+            "x86_64-unknown-uefi",
+            "x86_64-uwp-windows-gnu",
+            "x86_64-uwp-windows-msvc",
+            "x86_64-wrs-vxworks"
+        )
+    )
+
+    @Suppress("SameParameterValue")
+    private fun checkCompletion(
+        text: String,
+        expectedCompletions: List<String>
+    ) {
+        val testProject = buildProject {
+            toml("Cargo.toml", """
+                [package]
+                name = "hello"
+                version = "0.1.0"
+                authors = []
+            """)
+
+            dir("src") {
+                rust("main.rs", """
+                    fn main() {}
+                """)
+            }
+        }
+
+        val cargoProjects = testProject.psiFile("src/main.rs").project.cargoProjects
+        val cargoProject = cargoProjects.allProjects.first()
+        val provider = CargoCommandCompletionProvider(cargoProjects, cargoProject.workspace)
+        val (ctx, prefix) = provider.splitContextPrefix(text)
+        val matcher = PlainPrefixMatcher(prefix)
+
+        val actual = provider.complete(ctx).filter { matcher.isStartMatch(it) }.map { it.lookupString }
+        check(actual == expectedCompletions) {
```

Comment:
Let's not check expected values and actual by equality because actual list may be different for different toolchain versions. Let's check that some values are in actual values. For example, https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-1